classes/includes/src/Services/Gateway/HpsAttachmentService.php

```diff
@@ -54,14 +54,16 @@ public function getAttachments($GatewayTxnId, $AttachmentType = 'all', $ReturnAt
     {
         $GatewayTxnId = filter_var($GatewayTxnId, FILTER_SANITIZE_NUMBER_FLOAT);
         if (!$GatewayTxnId) {
-            throw new HpsArgumentException('Gateway Transaction ID required', HpsExceptionCodes::INVALID_NUMBER);
+            throw new HpsArgumentException('Gateway Transaction ID required', esc_html(HpsExceptionCodes::INVALID_NUMBER));
         }
 
         $AttachmentTypeProvided = preg_match(HpsAttachmentType::VALID_ATTACHMENT_TYPE, $AttachmentType) === 1;
         $AttachmentDataId = filter_var($AttachmentDataId, FILTER_SANITIZE_NUMBER_INT);
         // this is a limitation of the gateway and we want to inform the user
         if ($AttachmentTypeProvided && $AttachmentDataId) {
-            throw new HpsGatewayException(HpsExceptionCodes::GATEWAY_ERROR, "Since the AttachmentDataId was provided the AttachmentType was ignored by the server");
+            throw new HpsGatewayException(esc_html(HpsExceptionCodes::GATEWAY_ERROR),
+                "Since the AttachmentDataId was provided the AttachmentType was ignored by the server"
+            );
```

Comment:
```suggestion
            throw new HpsGatewayException(
                esc_html(HpsExceptionCodes::GATEWAY_ERROR),
                "Since the AttachmentDataId was provided the AttachmentType was ignored by the server"
            );
```
```suggestion
            throw new HpsGatewayException(esc_html(HpsExceptionCodes::GATEWAY_ERROR),
                "Since the AttachmentDataId was provided the AttachmentType was ignored by the server"
            );
```